js/super_hero.js

```diff
@@ -1 +1,182 @@
-// var SuperHero = ...
+// //SOLUTION PART 1
+
+// //Create SuperHero class
+// var SuperHero = function (superName) {
+// 	this.superName = superName;
+// 	this.hasSecretPower = true;
+// 	this.isAlive = true;
+// 	this.hasWeapon = true;
+// 	this.hasShield = true;
+// };
+
+// //Method using random number generator to determine 
+// //which properties are active on instance of SuperHero
+// SuperHero.prototype.powerAppraisal = function (){
+
+// 	if (parseInt(Math.random() * 5) % 2 === 0) {
+// 		this.hasSecretPower = false;
+// 	}
+
+// 	if (parseInt(Math.random() * 5) % 2 === 0) {
+// 		this.hasShield = false;
+// 	}
+
+// 	if (this.hasSecretPower === false && this.hasShield === true) {
+// 		return "My weapons will save me";}
+
+// 	if (this.hasShield === false && this.hasSecretPower === false ) {
+// 		return "Uh-oh. This is going to be sticky";}
+
+// else if (this.hasSecretPower === true && this.hasShield === true ||
+//  this.hasSecretPower === true && this.hasShield === false) {
+// 	return "You'll never Harm Me!!";
+// }
+// };
+
+// //List of active properties on superPerson instance of SuperHero
+// var superPerson = new SuperHero('superPerson');
+// console.log("Name: " + superPerson.superName);
+// console.log("Appraisal of power: " + superPerson.powerAppraisal());
+// console.log("Secret Power: " + superPerson.hasSecretPower);
+// console.log("Alive?: " + superPerson.isAlive);
+// console.log("Has Weapon?: " + superPerson.hasWeapon);
+// console.log("Has Shield?: " + superPerson.hasShield);
+
+
+ //---------------------------------------------------------
+ //SOLUTION PART 2 **** STRINGS REPLACED WITH VARIABLES ****
+ //---------------------------------------------------------
+ 
+ //Create SuperHero class
+ // "use strict";
+ var SuperHero = function (superName) {
+ 	this.superName = superName;
+ 	this.hasSecretPower = true;
+ 	this.isAlive = true;
+ 	this.hasWeapon = true;
+ 	this.hasShield = true;
+};
+
+function SuperChild(superName) {
+	SuperHero.call(this, superName);
+	this.hasBomb = true;
+}
+
+SuperChild.prototype = Object.create(SuperHero.prototype);
+SuperChild.prototype.constructor = SuperChild;
+
+
+// //Variables substitute for conversation strings as follow:
+var a = "My battle skills combined with these weapons will do you in!!";
+var b = "I will fight you with my bare hands!!";
+var c = "With these secret powers you don't have a chance!!";
+var d = "With this shield any weapon you try is just a toy!!";
+var e = "Ha! Your attack is no match for my secret powers!!";
+var f = "GOOD AND EVIL BATTLE TO THE END!!!";
+var g = "EVIL HAS TRIUMPHED -- BUT IT WILL NEVER ENDURE!!";
+var h = "GOOD TRUMPS EVIL AGAIN!!";
+var j = "NOT SO FAST! THIS BOMB IS THE END TO ALL YOUR EVIL";
+
+
+ 
+ //Method using random number generator to determine 
+ //which properties are active on instance of SuperHero
+ SuperHero.prototype.attack = function (){
+ 
+ 	if (Math.round(Math.random() * 18) % 2 === 0) {
```

Comment:
The logic is sound here, but it wouldn't be immediately obvious to a reader what it's actually checking. How would you describe the condition, in plain English? Then, make a method with that name, and it will be easier for the reader to follow. Something like `if (chance(2, 18)) { ...`.